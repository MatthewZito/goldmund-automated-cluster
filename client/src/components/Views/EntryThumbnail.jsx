import React from 'react';
import { NavLink } from "react-router-dom";

const EntryThumbnail = (props) => {
    return (
        <div className="grid-item">
            <img className="img-responsive" alt="" src={props.imgsrc} />
            <NavLink to={`/entry/${props.id}`} className="project-description">
                <div className="project-text-holder">
                <div className="project-text-inner">
                    <h3>{props.title}</h3>
                    <p>{props.subtitle}</p>
                </div>
                </div>
            </NavLink>
            </div>
    );
}
export default EntryThumbnail;