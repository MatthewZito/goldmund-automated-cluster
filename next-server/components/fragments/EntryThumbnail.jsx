import Link from "next/link";

const EntryThumbnail = (props) => {
    return (
        <div className="grid-item">
            <img className="img-responsive" alt="" src={props.imgsrc} />
            <Link href="/entry/[slug]" as={`/entry/${props.slug}`}>
                <a className="project-description">
                <div className="project-text-holder">
                    <div className="project-text-inner">
                        <h3>{props.title}</h3>
                        <p>{props.subtitle}</p>
                    </div>
                </div>
                </a>
            </Link>
        </div>
    );
}
export default EntryThumbnail;