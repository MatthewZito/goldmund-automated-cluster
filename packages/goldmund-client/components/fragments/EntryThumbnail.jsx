import React from 'react';
import Link from "next/link";

const EntryThumbnail = (props) => {
    return (
        <div key={props._id} className="grid-item" >
            <img className="img-responsive" alt="blog entry thumbnail" src={props.imgsrc} />
            <Link href="/entries/[slug]" as={`/entries/${props.slug}`}>
                <a className="project-description">
                <div className="project-text-holder">
                    <div className="project-text-inner">
                        <h3>{props.title}</h3>
                        <p>{props.subtitle}</p>
                    </div>
                </div>
                </a>
            </Link>
        </div>
    );
}
export default EntryThumbnail;