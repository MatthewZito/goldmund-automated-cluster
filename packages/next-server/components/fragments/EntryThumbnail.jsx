import Link from "next/link";

const EntryThumbnail = (props) => {
    return (
        <div key={props._id} className="grid-item" style={{display:"flex"}}>
            <img className="img-responsive" alt="blog entry image" src={props.imgsrc} />
            <Link href={`/entries/${props.slug}`}>
                <a className="project-description">
                <div className="project-text-holder">
                    <div className="project-text-inner">
                        <h3>{props.title}</h3>
                        <p>{props.subtitle}</p>
                    </div>
                </div>
                </a>
            </Link>
        </div>
    );
}
export default EntryThumbnail;