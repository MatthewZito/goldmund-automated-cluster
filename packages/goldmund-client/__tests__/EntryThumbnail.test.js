import { shallow } from "enzyme";
import React from 'react';
import EntryThumbnail from "../components/fragments/EntryThumbnail.jsx";

it("should render the EntryThumbnail component", () => {
    expect(shallow(<EntryThumbnail />).debug()).toMatchSnapshot();
});