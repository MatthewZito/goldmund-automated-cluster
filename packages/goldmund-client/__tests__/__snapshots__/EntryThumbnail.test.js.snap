// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the EntryThumbnail component 1`] = `
"<div className=\\"grid-item\\" style={{...}}>
  <img className=\\"img-responsive\\" alt=\\"blog entry thumbnail\\" src={[undefined]} />
  <Link href=\\"/entries/[slug]\\" as=\\"/entries/undefined\\">
    <a className=\\"project-description\\">
      <div className=\\"project-text-holder\\">
        <div className=\\"project-text-inner\\">
          <h3 />
          <p />
        </div>
      </div>
    </a>
  </Link>
</div>"
`;
